### PR TITLE
Handling of switch expressions and rule cases in Flow and NPECheck hint.

### PR DESCRIPTION
**Pull request by jlahoda**

### CODE REVIEW

File: java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java [hunk 1]
```diff
@@ -1172,37 +1207,63 @@ public Boolean visitBreak(BreakTree node, ConstructorData p) {
...
             
+            breakTo(target);
+
             return null;
         }
 
+        private void breakTo(Tree target) {
+            resumeAfter(target, variable2State);
+
+            variable2State = new HashMap< Element, State>();
```
Comment (jlahoda-jackpot) [inline java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java:1228]: Jackpot:
warning: Redundant type arguments in new expression (use diamond operator instead).

Comment (jlahoda) [inline java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java:1228]:
```suggestion
            variable2State = new HashMap<>();
```

File: java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java [hunk 2]
```diff
@@ -444,11 +445,45 @@ protected boolean isCanceled() {
...
+        }
+
         @Override
         public Boolean scan(Tree tree, ConstructorData p) {
             resume(tree, resumeBefore);
             
-            Boolean result = super.scan(tree, p);
+            Boolean result;
+
+            if (tree != null) {
```
Comment (sdedic) [inline java/java.hints/src/org/netbeans/modules/java/hints/introduce/Flow.java:470]: Wouldn't it be beneficial to build the `visitSwitchExpression` and `visitYield` into baseclasses ?

**Comment by lkishalmi** [comment]
@jlahoda @sdedic what is the state of this PR? Can it go in?

**Review by lkishalmi — APPROVED** [review]
Well skimming through the source code I found nothing that would cry for explanation.

**Pull request merged**